lib/core.mli

```diff
@@ -1,4 +1,4 @@
-type element = private Ojs.t
```

Comment:
If we remove this we "leak" the internal implementation of `element` and stops being abstract, right? I'm not sure we want to do this.

Reply:
Unfortunately, the way the ppx works with `Ojs.call` means that the input and output types both have to be the same, i.e.

```ocaml
external call: t -> string -> t array -> t = "caml_js_meth_call"
```

This means that `Imports.react` and `element` have to be the same type. There might be a way to get around this that I can look into, though

Reply:
Hm I see. I'm not sure exactly how, but if I restore the `private Ojs.t` definition, everything seems to keep working. 😕

Reply:
(to clarify, I would expect `Core` compilation to work fine as the internal implementation is visible inside the same module, but I would expect usages in `Dom` to fail, as they can't see the internal implementation of `element`)

Reply:
Oh you're right, that's very interesting. It might have been the way I was writing the bindings previously that produced the type error, and this way is fine (I'm also not sure why that would be, but `gen_js_api` uses `Obj.magic` in a lot of places, so maybe that's it)